[WIP] Visibility to pcj

isper3at opened the pull request:
Added decorator or BindingSets and Integration tests to account for the decorator and visibility

isper3at committed:
[WIP] Added visibility support for PCJ tables

Visibility support added for PCJ tables.
If no visibilities are defined during the query,
only binding sets visible to the accesssor will
be returned.

Added and updated tests for PCJ visibility.

isper3at commented:
@kchilton2 review me

kchilton2 commented on extras/indexing/src/test/java/mvm/rya/indexing/external/BindingSetDecorator.java:
```diff
@@ -0,0 +1,60 @@
+ */
+public abstract class BindingSetDecorator implements BindingSet {
+	private static final long serialVersionUID = 1L;
+	private BindingSet set;
```
Make final.

kchilton2 commented on extras/indexing/src/main/java/mvm/rya/indexing/accumulo/VisibilityBindingSet.java:
```diff
@@ -0,0 +1,52 @@
+/**
+ * Decorates a {@link BindingSet} with a collection of visibilities.
+ */
+public class VisibilityBindingSet extends BindingSetDecorator {
```
This class' functions need documentation.

kchilton2 commented on extras/indexing/src/main/java/mvm/rya/indexing/accumulo/VisibilityBindingSet.java:
```diff
@@ -0,0 +1,52 @@
+	}
+	
+	@Override
+	public boolean equals(Object o) {
```
This will throw a ClassCastException if o is not of type VisibilityBindingSet.

public boolean equals(Object o) {
    if(this == o) {
        return true;
    } else if(o instanceof VisibilityBindingSet) {
        // CHECK HERE
    }
    return false;
}

kchilton2 commented on extras/indexing/src/main/java/mvm/rya/indexing/accumulo/VisibilityBindingSet.java:
```diff
@@ -0,0 +1,52 @@
+ */
+public class VisibilityBindingSet extends BindingSetDecorator {
+	private static final long serialVersionUID = 1L;
+	private Collection<String> visibilitySet;
```
final

kchilton2 commented on extras/indexing/src/test/java/mvm/rya/indexing/external/AccumuloPcjVisibilityTest.java:
```diff
@@ -0,0 +1,275 @@
+import mvm.rya.rdftriplestore.RdfCloudTripleStore;
+import mvm.rya.rdftriplestore.RyaSailRepository;
+
+public class AccumuloPcjVisibilityTest {
```
Maybe decouple an integration test that requires an AccumuloMiniCluster from your actual tests. So move all the setup stuff into something called ITAccumuloBase and then have your actual integration tests extend it.

isper3at replied:
uhhh I think I was using this as a test bed for me.  This isn't actually a valid test

kchilton2 commented on extras/indexing/src/test/java/mvm/rya/indexing/external/tupleSet/PcjTablesIntegrationTests.java:
```diff
@@ -78,6 +79,7 @@
 import com.google.common.base.Optional;
 import com.google.common.collect.HashMultimap;
 import com.google.common.collect.Multimap;
```
This class could benefit from a ITAccumuloBase  as well.

kchilton2 commented on extras/indexing/src/test/java/mvm/rya/indexing/external/tupleSet/PcjTablesIntegrationTests.java:
```diff
@@ -335,12 +437,18 @@ public void createAndPopulatePcj() throws RepositoryException, PcjException, Tab
-            Scanner scanner = accumuloConn.createScanner(pcjTableName, new Authorizations());
+        	Authorizations authorizations;
+        	if(auths.length == 0) {
+        		authorizations = accumuloConn.securityOperations().getUserAuthorizations(accumuloConn.whoami());
```
You're using tab characters instead of 4 spaces.

isper3at replied:
yeah just noticed that.  fixing it now

kchilton2 commented on extras/indexing/src/main/java/mvm/rya/indexing/external/tupleSet/PcjTables.java:
```diff
@@ -630,7 +631,8 @@ private void writeResults(
                 // Row ID = binding set values, Column Family = variable order of the binding set.
                 Mutation addResult = new Mutation(serializedResult);
-                addResult.put(varOrder.toString(), "", "");
+                String visibility = "(" + String.join(")&(", result.getVisibility()) + ")";
```
Make this Java 7.

isper3at committed:
[WIP] Add visibility to fluo -- start

This does not allow the addition of visibilities to fluo,
it is merely a change to the insert and putting empty
visibility

kchilton2 commented on extras/rya.pcj.fluo/pcj.fluo.api/src/main/java/org/apache/rya/indexing/pcj/fluo/api/InsertTriples.java:
```diff
@@ -58,9 +63,12 @@
      */
-    public void insert(final FluoClient fluo, final RyaStatement triple) {
-        insert(fluo, Collections.singleton(triple));
+    public void insert(final FluoClient fluo, final RyaStatement triple, final byte[] visibility) {
```
String visibility instead of byte[]
